### ISSUE TITLE
editor.action.format blocks external formatter plugins

### ISSUE THREAD

**Issue by bskydive** [Microsoft/vscode#54346]
Issue Type: <b>Bug</b>

pugBeautify.execute can't run after error message "can't find registered formatter for jade files".

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:40:20.190Z)
OS version: Linux x64 4.4.138-59-default

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD A12-9720P RADEON R7, 12 COMPUTE CORES 4C+8G (4 x 1800)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>vpx_decode: unavailable_software<br>webgl: unavailable_off<br>webgl2: unavailable_off|
|Load (avg)|1, 1, 1|
|Memory (System)|7.27GB (0.39GB free)|
|Process Argv|/usr/share/code/code --unity-launch|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (16)</summary>

Extension|Author (truncated)|Version
---|---|---
overtype|ada|0.2.0
ng-template|Ang|0.1.9
bracket-pair-colorizer|Coe|1.0.59
tslint|eg2|1.0.33
copy-relative-path-and-line-numbers|ezf|0.2.8
beautify|Hoo|1.3.2
vscode-jetbrains-keybindings|isu|0.1.5
Angular2|joh|6.1.1
camelcasenavigation|map|0.3.48
vscode-pugbeautify|mrm|1.0.2
vscode-language-pack-ru|MS-|1.25.2
vscode-css-peek|pra|2.1.0
vscode-icons|rob|7.24.0
sass-indented|rob|1.4.9
code-settings-sync|Sha|3.0.0
language-stylus|sys|1.9.2

(1 theme extensions excluded)

</details>

**Comment by bskydive** [comment]
@Tyriar please, read ticket more attentive before labeling it. This issue caused **both** by VSCode and extension. In the description I'm wrote about VSCode bug, not about extension settings. Extension stop working because VSCode updates and demands extension refactoring. So, this is the VSCode side in both cases.

**Comment by jrieken** [comment]
@bskydive Please try without extensions enabled. I doubt this is an issue with VS Code, I could not find that error message ("can't find registered formatter for") in our sources.

**Comment by bskydive** [comment]
@jrieken thanks for response! 
It's my fault about caption. Were was incorrect translation.
This is the exact message with lang:en:
![default](https://user-images.githubusercontent.com/4169563/42927221-6e6f44ae-8b3c-11e8-8d5a-70d9bf7ace07.png)
Moreower, formatting with pugBeautify.execute works standalone, but didn't work in bunch with VSCode internal formatter, binded on the same keys.
![default](https://user-images.githubusercontent.com/4169563/42927599-a3b3283c-8b3d-11e8-9663-143d0ae848b8.png)
If I bind pugBeautify on the different keys or disable editor.action.format binding, it works fine. So, were is somefing wrong with formatting chaining, something stops the chain with error. Plugins and VSCode modules shouldn't interrupt each over.

Currently, it can work again after language switch, deleting and installing extension, removing and adding key bindings in json. I will describe this workaround in plugin repo. 

Please, check the module and plugins inferrence in simultaneous executing on the same key bindings.
Please, add the recomendation to reinstall plugins after refactoring file-attributes-binding depended code.

P.S.
Pug - is the only solution for stylus-like professional templating for html. It's reduce a lots of time on reading, writing and searching code. Plugins like https://marketplace.visualstudio.com/items?itemName=formulahendry.auto-rename-tag became useless with this templating approach. Please, help to keep it working in VSCode!

**Comment by bskydive** [comment]
I think, that order in list of keybindings take the main role. Now pug is the first, and later, it was the last, after editor.action.format, and hangs after previous error

**Comment by bskydive** [comment]
Not sure, that this issue must be closed, because bug can be still alive.

**Comment by jrieken** [comment]
@bskydive Just, to double check. It's this extension:https://marketplace.visualstudio.com/items?itemName=mrmlnc.vscode-pugbeautify? Iff so, the extension is to blame because it doesn't integrate into our formatter infrastructure but uses a separate command. We have blogged about this https://code.visualstudio.com/blogs/2016/11/15/formatters-best-practices and it's on the extension to integrate better.

**Comment by bskydive** [comment]
yes, it's that extension. And no, it need to stay work before somebody refactor it because it's only one plugin for that preprocessor. I have over 40 hours hard work during week, and abandoning simultaneous plugin execution interruption problem will grow this time. On the other side, I can't pay for fixing this problem, and wokraround exists.

Thanks for your time!

**Comment by bskydive** [comment]
An update: workaround makes editor.action.format broken. E.g. order in the key bindings is matter.

**Comment by jrieken** [comment]
@mrmlnc Any chance that you will update your extension?

**Comment by bskydive** [comment]
@mrmlnc It's not mine) Chanses is very low. I'm overloaded in this year. So, I will recomend to use separate key binding until it wirks.

**Comment by jrieken** [comment]
> So, I will recomend to use separate key binding until it works.

What different keybinding?

**Comment by bskydive** [comment]
Any different from keys assigned to editor.action.format

**Comment by jrieken** [comment]
I don't understand. Do you want us to make a change or do you want to assign some key binding to that custom command the extension contributed?

**Comment by bskydive** [comment]
Sorry for my english. I prefer to assign a different key to pug format, because chanses of extension refactoring is very low. Thanks for the link to best practices. I hopes, that can find a time for writing my own extension or refactor existing in this year.
Issue closed.

P.S.
Can you attach to the next VSCode release notes list of the extensions, that needs refactoring?
